.github/workflows/publish.yml

```diff
@@ -21,15 +21,15 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install setuptools wheel twine
+          pip install build twine
 
       - name: Build package
         run: |
-          python setup.py sdist bdist_wheel
+          python -m build
 
       - name: Publish package to PyPI
         env:
-          TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-          TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+          TWINE_USERNAME: "__token__"
```

Comment:
1) we could use gitaction in the market like https://github.com/pypa/gh-action-pypi-publish to publish.
2) The publish should be triggered either a) manual b) a new release tag has been made. Not everytime main changed.